pkg/backup/backup_test.go

```diff
@@ -53,6 +53,54 @@ import (
 	"github.com/heptio/velero/pkg/volume"
 )
 
+func TestBackedUpItemsMatchesTarballContents(t *testing.T) {
+	// TODO: figure out if this can be replaced with the restmapper
+	// (https://github.com/kubernetes/apimachinery/blob/035e418f1ad9b6da47c4e01906a0cfe32f4ee2e7/pkg/api/meta/restmapper.go)
+	gvkToResource := map[string]string{
+		"v1/Pod":              "pods",
+		"apps/v1/Deployment":  "deployments.apps",
+		"v1/PersistentVolume": "persistentvolumes",
+	}
+
+	h := newHarness(t)
+	req := &Request{Backup: defaultBackup().Result()}
+	backupFile := bytes.NewBuffer([]byte{})
+
+	apiResources := []*test.APIResource{
+		test.Pods(
+			builder.ForPod("foo", "bar").Result(),
+			builder.ForPod("zoo", "raz").Result(),
+		),
+		test.Deployments(
+			builder.ForDeployment("foo", "bar").Result(),
+			builder.ForDeployment("zoo", "raz").Result(),
+		),
+		test.PVs(
+			builder.ForPersistentVolume("bar").Result(),
+			builder.ForPersistentVolume("baz").Result(),
+		),
+	}
+	for _, resource := range apiResources {
+		h.addItems(t, resource)
+	}
+
+	h.backupper.Backup(h.log, req, backupFile, nil, nil)
+
+	var expectedFiles []string
+	for item := range req.BackedUpItems {
```

Comment:
Another way would be to just hardcode the expected BackedUpItems and filepaths and assert that they match?

Reply:
This works for me - let's just add a comment that explains the approach here (ensure BackedUpItems matches the tarball contents by looking at BackedUpItems after backup to assemble a list of files expected to be in the tarball, then comparing).

Reply:
:+1: added!